packages/x-components/src/x-modules/scroll/wiring.ts

```diff
@@ -26,9 +26,22 @@ export const setScrollPositionWire = wireCommit(
  * @public
  */
 export const setScrollDirectionWire = wireCommit(
-  'setScrollDirection',
+  'setScrollComponentState',
   ({ metadata, eventPayload }: WirePayload<ScrollDirection>) => ({
-    direction: eventPayload,
+    newState: { direction: eventPayload },
+    id: metadata.id as string
```

Comment:
Can you use `id: metadata.id!` here? At this point you can asume that you're going to have and id right?